index.js

```diff
@@ -32,105 +33,188 @@ function loader (registryOrVersion) {
-            item1.metadata === item2.metadata &&
-            JSON.stringify(item1.nbt) === JSON.stringify(item2.nbt))
+        return (
+          item1.type === item2.type &&
+          (matchStackSize ? item1.count === item2.count : true) &&
+          item1.metadata === item2.metadata &&
+          (matchNbt ? JSON.stringify(item1.nbt) === JSON.stringify(item2.nbt) : true)
+        )
       }
     }
 
-    static toNotch (item) {
+    // Stack ID
+    static currentStackId = 0
+    static nextStackId () {
+      return Item.currentStackId++
+    }
+
+    static toNotch (item, serverAuthoritative = true) {
       if (registry.supportFeature('itemSerializationAllowsPresent')) {
         if (item == null) return { present: false }
-        const notchItem = {
+        const networkItem = {
           present: true,
           itemId: item.type,
           itemCount: item.count
         }
-        if (item.nbt && item.nbt.length !== 0) { notchItem.nbtData = item.nbt }
-        return notchItem
+        if (item.nbt && item.nbt.length !== 0) {
+          networkItem.nbtData = item.nbt
+        }
+        return networkItem
       } else if (registry.supportFeature('itemSerializationUsesBlockId')) {
         if (item == null) return { blockId: -1 }
-        const notchItem = {
+        const networkItem = {
           blockId: item.type,
           itemCount: item.count,
           itemDamage: item.metadata
         }
-        if (item.nbt && item.nbt.length !== 0) { notchItem.nbtData = item.nbt }
-        return notchItem
+        if (item.nbt && item.nbt.length !== 0) {
+          networkItem.nbtData = item.nbt
+        }
+        return networkItem
+      } else if (registry.type === 'bedrock') {
+        // This can later be changed to use supportFeature
+        if (item.type === 0) return { network_id: 0 }
+        if (registry.version['<']('1.16.220')) {
+          const networkItem = {
+            network_id: item.id,
+            auxiliary_value: (item.metadata << 8) | (item.count & 0xff),
+            has_nbt: item.nbt !== null,
+            nbt: item.nbt !== null ? { version: 1, nbt: item.nbt } : undefined,
+            can_place_on: item.blocksCanPlaceOn,
+            can_destroy: item.blocksCanDestroy
+            // blocking_tick: 0, // TODO
+          }
+
+          return networkItem
+        } else {
+          const networkItem = {
+            network_id: item.type,
+            count: item.count,
+            metadata: item.metadata,
+            has_stack_id: +serverAuthoritative,
+            stack_id: serverAuthoritative ? item.stackId : undefined,
+            block_runtime_id: 0, // TODO
+            extra: {
+              has_nbt: item.nbt !== null,
+              nbt: item.nbt !== null ? { version: 1, nbt: item.nbt } : undefined,
+              can_place_on: item.canPlaceOn,
+              can_destroy: item.canDestroy
+              // blocking_tick: 0, // TODO
```

Comment:
Set to 0

Reply:
If I understand the [docs](https://minecraft-data.prismarine.js.org/?v=bedrock_1.19.62&d=protocol#Item) correctly it's present if the item is the shield and absent if it isn't? That should be easy enough to test for

Reply:
It should be set to 0 at all times, the serializer will take care of writing it or not